runtime/lua/vim/diagnostic.lua

```diff
@@ -895,22 +895,30 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = vim.api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    local sign_ns = ns.user_data.sign_ns
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_group, sign_highlight_map[diagnostic.severity], bufnr, {
+      local lnum = diagnostic.lnum
+      local col = diagnostic.col
+      local sign_text = opts.sign_text and opts.sign_text[diagnostic.severity] or M.severity[diagnostic.severity]:sub(1,1)
+      local sign_opts = {
+        sign_text = sign_text,
+        sign_hl_group = sign_highlight_map[diagnostic.severity],
         priority = get_priority(diagnostic.severity),
-        lnum = diagnostic.lnum + 1,
-      })
+      }
+      pcall(vim.api.nvim_buf_set_extmark, bufnr, sign_ns, lnum, col, sign_opts)
```

Comment:
Why the `pcall`?

Reply:
It is probably unnecessary, but I was occasionally getting an error from column index. I suppose I could just set col to be 0 since the location in the line of the diagnostic is somewhat irrelevant for a sign.

Reply:
Definitely don't want a `pcall` here. If something can fail, we need to guard against it explicitly.

Reply:
> Definitely don't want a `pcall` here. If something can fail, we need to guard against it explicitly.

I did some testing and the error occurs when a line is deleted. For example if there is an error on line 10, and I delete line 10, with `dd` from normal mode, it will throw a col outside range error. This is just an issue with extmarks in general I think, and I'm not sure how to guard against it explicitly without just setting the column to 0 (as opposed to the starting point of the actual diagnostic) or setting strict to off (which is functionally the same as a pcall in this scenario)

Reply:
My vote would be on `col=0` then, as signs don't care about columns.

(Strict is not equivalent to pcall, either semantically or functionally.)

Reply:
> My vote would be on `col=0` then, as signs don't care about columns.
> 
> (Strict is not equivalent to pcall, either semantically or functionally.)

I phrased that very poorly, I just meant that the error it avoids is the same one I used pcall to ignore, but since other errors would be suppressed by pcall that would not be by strict, I agree that it is functionally different.

I also think col being 0 is the best choice here though.